File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/operation/meta/value/factory/LongValueFactory.java
package ru.citeck.ecos.records3.record.operation.meta.value.factory;

import ru.citeck.ecos.records3.record.operation.meta.value.AttValue;

import java.util.Collections;
import java.util.List;

public class LongValueFactory implements AttValueFactory<Long> {

    @Override
    public AttValue getValue(Long value) {
        return new AttValue() {
            @Override
            public String getString() {
                return value.toString();
            }

            @Override
            public Double getDouble() {
                return Double.valueOf(value);
            }

            @Override
            public Boolean getBool() {
                return value != 0;
            }
        };
    }

    @Override
    public List<Class<? extends Long>> getValueTypes() {
        return Collections.singletonList(Long.class);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/factory/LongValueFactory.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.factory

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue

class LongValueFactory : AttValueFactory<Long> {

    override fun getValue(value: Long): AttValue? {

        return object : AttValue {

            override fun asText(): String? {
                return value.toString()
            }

            override fun asDouble(): Double? {
                return value.toDouble()
            }

            override fun asBoolean(): Boolean? {
                return value != 0L
            }
        }
    }

    override fun getValueTypes() = listOf(Long::class.java)
}
